File: pset1/dinheiro/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void){
    float reais = get_float("Troco necessário em reais?");
    while (reais < 0){
        reais = get_float("Troco necessário em reais?");
    }

    float centavos = round(reais * 100);

    printf("%f\n", centavos);

    int moedas = 0;

    while (centavos > 0){
        if (centavos >= 25){
            centavos -= 25;
            moedas++;
        }else if (centavos >= 10){
            centavos -= 10;
            moedas++;
        }else if (centavos >= 5){
            centavos -= 5;
            moedas++;
        }else{
            centavos--;
            moedas++;
        }

    }


    printf("%f\n", centavos);
    printf("%i\n", moedas);
}

File: pset6/cash/cash.py
import cs50

def main():
    while True:
        valor = cs50.get_float("Valor: ")
        if valor > 0:
            break
    
    valor = round(valor*100)
    moedas = 0 
    moeda = (25,10,5,1)
    i = 0
    for i in range(0, 4, 1):
        while valor >= moeda[i]:
            valor -= moeda[i]
            moedas += 1
    
    print(moedas)

main()